### DISCUSSION TITLE
Upgrading to 7.1.1 from 7.1.0 with Macports

### DISCUSSION THREAD

**Discussion by memphis-10** [ImageMagick/ImageMagick#6563] [Help]
### ImageMagick version

7.1.0

### Operating system, version and so on

Mac Mojave

### Description

Hi. I keep an instance of Mac Mojave running for various reasons and would like to upgrade IM to the latest version.  Homebrew doesn't like Mojave and errors before getting to IM upgrade, so I am curious if anyone can offer advice for using Macports (or another means) to upgrade?   I'm somewhat familiar with terminal installs, but I can't seem to find a solution.

Thank you

**Comment by fmw42** [comment]
I have used Mac OSX Sierra and now Ventura.  I install all my delegates with MacPorts and then install IM from source.

**Reply by memphis-10** [reply]
Thank you. Do you install delegates one by one?

**Reply by fmw42** [reply]
Yes, because I look to make sure there are no error messages before going to the next one.  For Imagemagick, you do not need that many so it should not take too long.  Too bad MacPorts only offers IM 6. So for IM 7, you would have to install from source (as I do). In fact I install both IM 6 and IM 7 from source. But that is another layer of detail.  Stick with IM 7 for now.

See for example:

http://www.imagemagick.org/discourse-server/viewtopic.php?f=1&t=29100

http://www.imagemagick.org/discourse-server/viewtopic.php?f=1&t=21502&p=88202&hilit=OSX+MacPorts#p88202

**Reply by memphis-10** [reply]
Thank you. I'll read those a couple times to be sure I understand.  Since I am going from 7.1.0 to 7.1.1, could I assume I have all the necessary delegates?

**Comment by fmw42** [comment]
Were they installed by MacPorts?  If not install via MacPorts. Or if they are old upgrade via MacPorts.

What delegates do you have installed?

**Reply by fmw42** [reply]
If you have not installed from source before, the issue will likely be finding the right compilers from MacPorts.  You may have to try different ones until you find one that works.  That was my experience.  Sorry, I did not do it for Mojave

**Reply by memphis-10** [reply]
Thank you. I installed IM 7.1.0 from Homebrew.  These are the delegates I have installed: bzlib fontconfig freetype gslib heic jng jp2 jpeg lcms lqr ltdl lzma openexr png ps tiff webp xml zlib

I found this link which has info for upgrading via Macports, but I'm not familiar enough with Macports to get the local repository to work for me

https://trac.macports.org/ticket/51310#comment:61

**Reply by fmw42** [reply]
You cannot use the delegates installed from Homebrew with MacPorts.  They need to be in different locations.  So you need to install MacPorts, then install each of those delegates using MacPorts.  Or Install a compiler. Then install IM from source pointing to the location of your compiler and delegates. For example the last config command that I used (I think it was for Sierra) was



```
./configure CC=/usr/bin/clang CXX=/usr/bin/clang++ \
CPPFLAGS='-I/opt/local/include' LDFLAGS='-L/opt/local/lib' \
--enable-openmp  \
--enable-delegate-build --enable-shared --disable-static --disable-opencl \
--with-modules --with-quantum-depth=16 --without-wmf --with-rsvg --with-fftw \
--disable-silent-rules --disable-dependency-tracking --without-pango \
--with-lqr --with-gslib --with-gs-font-dir=/opt/local/share/ghostscript/fonts/
```

(My compiler, clang/clang++, was in /usr/bin, but my MacPorts delegates were installed in /opt/local), 

then

```
make clean
make
sudo make install
```

**Comment by memphis-10** [comment]
Thank you. This mostly makes sense. I’ll try this

**Reply by fmw42** [reply]
Sorry, I am not a software expert. So this was just from experience. You can find more information at https://imagemagick.org/script/advanced-linux-installation.php#build
